app/uk/gov/hmrc/paymentsemailverification/repositories/EmailVerificationStatusRepo.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.paymentsemailverification.repositories
+
+import org.mongodb.scala.bson.BsonDocument
+import org.mongodb.scala.model._
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.Codecs
+import uk.gov.hmrc.paymentsemailverification.config.AppConfig
+import uk.gov.hmrc.paymentsemailverification.crypto.CryptoFormat.OperationalCryptoFormat
+import uk.gov.hmrc.paymentsemailverification.models.{CorrelationId, EmailVerificationStatus, GGCredId}
+import uk.gov.hmrc.paymentsemailverification.repositories.Repo.{Id, IdExtractor}
+import uk.gov.hmrc.paymentsemailverification.repositories.EmailVerificationStatusRepo._
+
+import java.util.concurrent.TimeUnit
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+final class EmailVerificationStatusRepo @Inject() (
+    mongoComponent: MongoComponent,
+    config:         AppConfig
+)(implicit ec: ExecutionContext, cryptoFormat: OperationalCryptoFormat)
+  extends Repo[CorrelationId, EmailVerificationStatus](
+    collectionName = "emailVerificationStatus",
+    mongoComponent = mongoComponent,
+    indexes        = EmailVerificationStatusRepo.indexes(config.emailVerificationStatusRepoTtl.toSeconds),
+    extraCodecs    = Codecs.playFormatCodecsBuilder(EmailVerificationStatus.format).build,
+    replaceIndexes = true
+  ) {
+
+  def findAllEntries(ggCredId: GGCredId): Future[List[EmailVerificationStatus]] =
+    collection
+      .find(filter = Filters.eq("credId", ggCredId.value))
+      .sort(BsonDocument("createdAt" -> -1))
+      .toFuture().map(_.toList)
+
+  def update(emailVerificationStatus: EmailVerificationStatus): Future[Unit] = upsert(emailVerificationStatus)
+}
+
+object EmailVerificationStatusRepo {
+
+  /**
```

Comment:
Will using correlationId make sense if this is being made generic? I'm not fussed now though, more of a thing to consider

Reply:
it's an internal thing so callers of the service don't need to worry about it. I seemed like I needed to give some sort of _id so I just went with this - not sure if there's a better name for it? 🤔

Reply:
I suck at names, so tbh correlation id kind of makes sense actually